src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -40,7 +40,10 @@ class TaskReplaceActor(
   // Killed resident tasks are not expunged from the instances list. Ignore
   // them. LaunchQueue takes care of launching instances against reservations
   // first
-  val currentInstances = instanceTracker.specInstancesSync(runSpec.id).filter(_.state.goal == Goal.Running)
+  val currentInstances = {
+    // TODO(karsten): Use InstanceTrackerConfig.internalTaskTrackerRequestTimeout or rather be async.
+    Await.result(scheduler.getInstances(runSpec.id), 1000.millis).filter(_.state.goal == Goal.Running)
```

Comment:
Where are those magical `1000.millis` coming from?

Reply:
Like the _TODO_ says, I'd rather move this code to the _async_ part of the `preStart` method and have it asynchronous.

Reply:
I'm inclined to just keep the specInstancesSync call here for now, since it seems not entirely necessary to change the sync nature of this approach as part of this PR.